### PR TITLE
Add yaml to H header convertor

### PR DESCRIPTION
**Pull request by AzulRadio**
Test
```
cd vehicles/config
python generate_all.py yaml_list
```
Expected behavior: generate all yaml files listed in yaml_list to h headers.

The idea is write different yaml file in `config_files/` and generate desired combination with different `yaml_list`.

### CODE REVIEW

**Comment by 0-EricZhou-0** [comment]
Maybe merge with main? Main branch is updated a while ago. 
I will be working on integrating the generation into makefiles and also usage of header.

File: vehicles/config/generate_all.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+'''
+
+# USER PARAMETERS START //
+
+# output dir for the generated .h file
+# if empty, use the same dir as the given YAML file
+output_dir = "./"
+
+# dictionary dir, convert str in YAML to int in .H
+dict_dir = "./dictionary.yaml"
```
Comment (0-EricZhou-0) [inline vehicles/config/generate_all.py:38]: Is it possible that these can also be passed in as command line arguments?

**Comment by AzulRadio** [comment]
@0-EricZhou-0 
> I will be working on integrating the generation into makefiles and also usage of header. 

I will be working on the Cmake integrating as well with another upcoming PR so you don't need to worry about that. 

>Also I think this config folder should be moved outside of vehicles

`/config` in root dir usually means the configuration for the entire project, which in our case it is not. I am still thinking about where to put it.

File: vehicles/config/generate_all.py (hunk 1 above)
Comment (AzulRadio) [reply vehicles/config/generate_all.py:38]: There is no need. `output_dir` and the `dict_dir` never change as soon as we decide them.

File: vehicles/config/config_files/yaml_to_header_convertor.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    
+    # load dictionary
+    if dict_dir != "":
+        dict_dir = dict_dir
+        dict_file = open(dict_dir, "r")
+        dictionary = yaml.safe_load(dict_file)
+        
+    # discard the .yaml part from the YAML file to get the dir
+    h_file_name = yaml_dir.split("/")[-1].split(".")
+    if len(h_file_name) == 1:
```
Comment (ngao4) [inline vehicles/config/config_files/yaml_to_header_convertor.py:44]: ...

**Review by ngao4 — APPROVED** [review]

**Review by ngao4 — APPROVED** [review]

**Pull request merged**